scan.js

```diff
@@ -1,80 +1,108 @@
-        .sc .backdoor > a {cursor:pointer; text-decoration:underline;}
-        .sc .cct {color:#0ff;}
-    </style>`;
-let doc = eval("document");
-let tprint = html => doc.getElementById("terminal").insertAdjacentHTML('beforeend', `<li>${html}</li>`);
-/** @param {NS} ns **/
-export let main = ns => {
-    let tIn = doc.getElementById("terminal-input");
-    let tEv = tIn[Object.keys(tIn)[1]];
-    let priorCss = doc.getElementById("scanCSS");
-    if (priorCss) priorCss.parentNode.removeChild(priorCss); // Remove old CSS to facilitate tweaking css above
-    doc.head.insertAdjacentHTML('beforeend', css);
-    let serverInfo = (x) => {
-        return ns.getServer(x); // Costs 2 GB. If you can't don't need backdoor links, uncomment and use the alternate implementations below
-        /* return {
-            requiredHackingSkill: ns.getServerRequiredHackingLevel(x),
-            hasAdminRights: ns.hasRootAccess(x),
-            purchasedByPlayer: x.includes('daemon') || x.includes('hacknet'),
-            backdoorInstalled: true // No way of knowing without ns.getServer
-        } */
-    }
-    let s = ["home"],
-        p = [""],
-        r = { home: "home" },
-        myHack = ns.getHackingLevel(),
-        fName = x => {
-            let server = serverInfo(x); // Costs 2 GB. If you can't don't need backdoor links, uncomment the alternate implementations below
-            let reqHack = server.requiredHackingSkill; // ns.getServerRequiredHackingLevel(x);
-            let rooted = server.hasAdminRights; // ns.hasRootAccess(x);
-            let shouldBackdoor = !server?.backdoorInstalled && reqHack <= myHack && x != 'home' && rooted && !server.purchasedByPlayer;
-            let contracts = ns.ls(x, ".cct"); // Find out whether there are any contracts on the server
-            return `<span class="w" id="${x}">` +
-                `<a class="s${factionServers.includes(x) ? " f" : ""}${rooted ? " r" : ""}">${x}</a>
-                <span class="hack ${(reqHack <= myHack ? 'green' : 'red')}">(${reqHack})</span>
-                ${(shouldBackdoor ? '<span class="backdoor">[<a>backdoor</a>]</span>' : '')}
-                ${contracts.map(c => `<span class="cct" title="${c}">@</span>`)}
-            </span>`;
-        };
-    let tcommand = x => {
-        tIn.value = x;
-        tEv.onChange({ target: tIn });
-        tEv.onKeyDown({ keyCode: "13", preventDefault: () => 0 });
-    };
+/**
+ * @param {NS} ns
+ * @returns interactive server map
+ */
+export function main(ns) {
+    const factionServers = ["CSEC", "avmnite-02h", "I.I.I.I", "run4theh111z", "w0r1d_d43m0n", "fulcrumassets"],
+        css = `    <style id="scanCSS">
+        .serverscan {white-space:pre; color:#ccc; font:14px monospace; line-height: 16px; }
+        .serverscan .server {color:#080;cursor:pointer;text-decoration:underline}
+        .serverscan .faction {color:#088}
+        .serverscan .rooted {color:#6f3}
+        .serverscan .rooted.faction {color:#688}
```

Comment:
Copy paste error?
```suggestion
        .serverscan .rooted.faction {color:#0ff}
```
Colour of rooted servers is meant to be "brighter". Here was meant to be #0ff (previously `.sc .r.f {color:#0ff})`:
![image](https://user-images.githubusercontent.com/2285037/234069377-26dd7664-4162-4824-9c0a-e6841fb31e68.png)
rather than #688:
![image](https://user-images.githubusercontent.com/2285037/234069495-0175aec8-d401-456e-97b0-2f9f30d5b5a8.png)